src/scripts/towns/Town.ts

```diff
@@ -215,34 +215,34 @@ const BattleFrontierShop = new Shop([
-TownList['Slateport City'] = new Town('Slateport City', [new ClearDungeonRequirement(1, Statistics.getDungeonIndex('Granite Cave'))]);
+TownList['Rustboro City'] = new Town('Rustboro City', [new ClearDungeonRequirement(1, Statistics.getDungeonIndex('Petalburg Woods'))]);
+TownList['Dewford Town'] = new Town('Dewford Town', [new ClearDungeonRequirement(1, Statistics.getDungeonIndex('Rusturf Tunnel'))]);
+TownList['Slateport City'] = new Town('Slateport City', [new ClearDungeonRequirement(1, Statistics.getDungeonIndex('Granite Cave')), new GymBadgeRequirement(BadgeCase.Badge.Knuckle)]);
 TownList['Mauville City'] = new Town('Mauville City', [new RouteKillRequirement(10, 110)]);
 TownList['Verdanturf Town'] = new Town('Verdanturf Town', [new RouteKillRequirement(10, 117)]);
-TownList['Lavaridge Town'] = new Town('Lavaridge Town', [new RouteKillRequirement(10, 115), new ClearDungeonRequirement(1, Statistics.getDungeonIndex('Mt. Chimney'))]);
+TownList['Lavaridge Town'] = new Town('Lavaridge Town', [new ClearDungeonRequirement(1, Statistics.getDungeonIndex('Jagged Pass'))]);
 TownList['Fallarbor Town'] = new Town('Fallarbor Town', [new RouteKillRequirement(10, 113)]);
 TownList['Fortree City'] = new Town('Fortree City', [new RouteKillRequirement(10, 119)]);
-TownList['LilyCove City'] = new Town('LilyCove City', [new RouteKillRequirement(10, 121), new ClearDungeonRequirement(1, Statistics.getDungeonIndex('Mt. Pyre'))], LilyCoveCityShop);
-TownList['Mossdeep City'] = new Town('Mossdeep City', [new RouteKillRequirement(10, 125), new ClearDungeonRequirement(1, Statistics.getDungeonIndex('Shoal Cave'))], MossdeepCityShop);
-TownList['Sootopolis City'] = new Town('Sootopolis City', [new RouteKillRequirement(10, 126), new ClearDungeonRequirement(1, Statistics.getDungeonIndex('Cave of Origin'))]);
-TownList['Ever Grande City'] = new Town('Ever Grande City', [new RouteKillRequirement(10, 128)]);
+TownList['LilyCove City'] = new Town('LilyCove City', [new RouteKillRequirement(10, 121)], LilyCoveCityShop);
+TownList['Mossdeep City'] = new Town('Mossdeep City', [new RouteKillRequirement(10, 125)], MossdeepCityShop);
+TownList['Sootopolis City'] = new Town('Sootopolis City', [new RouteKillRequirement(10, 126), new GymBadgeRequirement(BadgeCase.Badge.Mind)]);
+TownList['Ever Grande City'] = new Town('Ever Grande City', [new GymBadgeRequirement(BadgeCase.Badge.Rain)]);
 TownList['Pokemon League Hoenn'] = new Town('Pokemon League', [new RouteKillRequirement(10, 128), new ClearDungeonRequirement(1, Statistics.getDungeonIndex('Victory Road Hoenn'))]);
 TownList['Pacifidlog Town'] = new Town('Pacifidlog Town', [new RouteKillRequirement(10, 131)]);
 TownList['Battle Frontier'] = new Town('Battle Frontier', [new ClearDungeonRequirement(1, Statistics.getDungeonIndex('Victory Road Hoenn'))], BattleFrontierShop);
```

Comment:
This is not right, Frontier is post-League

Reply:
```suggestion
TownList['Battle Frontier'] = new Town('Battle Frontier', [new GymBadgeRequirement(BadgeCase.Badge.Elite_HoennChampion)], BattleFrontierShop);
```